python-avd/pyavd/_eos_designs/structured_config/network_services/utils_wan.py

```diff
@@ -37,184 +37,325 @@ def _filtered_wan_vrfs(self: AvdStructuredConfigNetworkServicesProtocol) -> EosD
 
         return wan_vrfs
 
-    @cached_property
-    def _wan_virtual_topologies_policies(self: AvdStructuredConfigNetworkServicesProtocol) -> EosDesigns.WanVirtualTopologies.Policies:
-        """This function parses the input data and append the default-policy if not already present."""
-        # If not overwritten, inject the default policy in case it is required for one of the VRFs
-        if self._default_wan_policy_name in self.inputs.wan_virtual_topologies.policies:
-            return self.inputs.wan_virtual_topologies.policies
+    @structured_config_contributor
+    def set_wan_policies(self: AvdStructuredConfigNetworkServicesProtocol) -> None:
+        if not self.shared_utils.is_wan_router:
+            return
 
-        policies = self.inputs.wan_virtual_topologies.policies._deepcopy()
-        policies.append(self._default_wan_policy)
-        return policies
+        append_policy = self._append_autovpn_policy if self.inputs.wan_mode == "autovpn" else self._append_cv_pathfinder_policy
```

Comment:
This would be more readable if you just do the `if`/`else` where you call the actual function (line 61)